src/utils.rs

```diff
@@ -45,29 +47,62 @@ impl StableHash for AsInt<'_> {
     }
 }
 
-pub fn stable_hash<T: StableHasher, S: SequenceNumber, V: StableHash>(value: &V) -> T::Out {
-    let mut hasher = T::default();
-    value.stable_hash(S::root(), &mut hasher);
+pub fn stable_hash<H: StableHasher + Default, T: StableHash>(value: &T) -> H::Out {
+    let mut hasher = H::default();
+    value.stable_hash(H::Seq::root(), &mut hasher);
     hasher.finish()
 }
 
 pub fn stable_hash_with_hasher<T: std::hash::Hasher + Default, V: StableHash>(value: &V) -> u64 {
-    stable_hash::<StableHasherWrapper<T>, SequenceNumberInt<u64>, _>(value)
+    stable_hash::<StableHasherWrapper<T, SequenceNumberInt<u64>>, _>(value)
 }
 
 /// Wraps a Hasher to implement StableHasher. It must be known that the Hasher behaves in
 /// a consistent manner regardless of platform or process.
 #[derive(Default)]
-pub struct StableHasherWrapper<T>(T);
+pub struct StableHasherWrapper<H, Seq> {
+    hasher: H,
+    _marker: PhantomData<*const Seq>,
+}
+
+pub struct XorAggregator<T> {
+    value: u64,
+    _marker: std::marker::PhantomData<*const T>,
```

Comment:
Why not `PhantomData<T>` here? Also `PhantomData` is imported so you don't need to qualify it.

Reply:
`PhantomData<T>` would infer that `XorAggregator` logically "owns" a `T`. But, the code only uses `T` to construct ephemeral values.

From the [docs for PhantomData](https://doc.rust-lang.org/std/marker/struct.PhantomData.html):

> If your struct does not in fact own the data of type T, it is better to use a reference type, like PhantomData<&'a T> (ideally) or PhantomData<*const T> (if no lifetime applies), so as not to indicate ownership.

Here, `T` does not factor into the drop check analysis. Some methods construct a `T` on the stack and use the value temporarily, but no ownership or lifetimes apply to the "containing" struct.

Reply:
Interesting, I went into this obscure corner of the language and that doc may be outdated https://github.com/rust-lang/rust/issues/70841 and there is a controversy I'm not even close to understanding around it. Anyways just a curiosity, nothing wrong with following the current docs here.